src/main/scala/PatternGenerator.scala

```diff
@@ -236,14 +239,20 @@ class PatternGenerator(dataWidth: Int, lineWidth: Int, samples: Int,
     nextState := sIdle
     started := started || combinationalTriggerFire
   } .elsewhen (state === sIdle && io.control.bits.arm && io.control.valid) {
-    nextState := sArmed
+    if (combinationalTrigger) {
+      nextState := sArming
+    } else {
+      nextState := sArmed
+    }
     confReadyBypass := io.control.bits.readyBypass
     confTriggerMode := io.control.bits.triggerMode
     confLastSample := io.control.bits.lastSample
     confContinuous := io.control.bits.continuous
     currSample := 0.U
     overflow := false.B
     started := false.B
+  } .elsewhen (state === sArming) {
+    nextState := sArmed
```

Comment:
Usually the person who opens the PR presses the button (in case there is still something left, etc), so I'll merge it.